hooks/checkout

```diff
@@ -8,8 +8,25 @@ if [[ "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_SKIP_CHECKOUT" == "true" ]]; then
     exit 0
 fi
 
-WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
 echo "--- :open_file_folder: Setting up workspace"
+WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
 echo "Creating directory: $WORKSPACE"
 mkdir -p "$WORKSPACE" && cd "$WORKSPACE"
 echo "Setup completed successfully."
+
+echo "--- Checking out repository"
+echo "Setting up git and ssh..."
+# setup git to use ssh instead of https
+git config --global url."ssh://git@github.com".insteadOf "https://github.com"
+mkdir -p "$HOME/.ssh" && ssh-keyscan -t rsa github.com >> "$HOME/.ssh/known_hosts"
+
+CLONE_URL=${BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_CLONE_URL:-$BUILDKITE_REPO}
+echo "Cloning repository: $CLONE_URL"
+git clone --no-checkout "$CLONE_URL" && cd "$(echo "${CLONE_URL%/}" | awk -F '/' '{ print $NF }')"
```

Comment:
Trying to understand this shell substitution `${CLONE_URL%/}`. I guess it takes the suffix after a `/` in the given env, right?

(any links that would help me understand is much appreciated here)

Reply:
This is for removing a trailing slash, if it exists, so that we can get the repo name by printing the last field separated by `/` using `awk`

Reference to parameter expansion in the bash manual [here](https://www.gnu.org/software/bash/manual/bash.html#Brace-Expansion)
> ${parameter%word}
${parameter%%word}
... If the pattern matches a trailing portion of the expanded value of parameter, then the result of the expansion is the value of parameter with the shortest matching pattern (the ‘%’ case) or the longest matching pattern (the ‘%%’ case) deleted. ...

Reply:
Thanks!